File: app/src/main/java/edu/aku/hassannaqvi/naunehal/models/UCs.java
package edu.aku.hassannaqvi.naunehal.models;


import android.database.Cursor;

import org.json.JSONException;
import org.json.JSONObject;

import edu.aku.hassannaqvi.naunehal.contracts.UCsContract;
import edu.aku.hassannaqvi.naunehal.contracts.UCsContract.TableUCs;

public class UCs {

    private static final String TAG = "UCs_CONTRACT";
    String ucCode;
    String ucName;
    String districtCode;

    public UCs() {
        // Default Constructor
    }

    public UCs Sync(JSONObject jsonObject) throws JSONException {
        this.ucCode = jsonObject.getString(TableUCs.COLUMN_UC_CODE);
        this.ucName = jsonObject.getString(TableUCs.COLUMN_UC_NAME);
        this.districtCode = jsonObject.getString(TableUCs.COLUMN_DISTRICT_CODE);
        return this;
    }

    public UCs HydrateUCs(Cursor cursor) {
        this.ucCode = cursor.getString(cursor.getColumnIndex(TableUCs.COLUMN_UC_CODE));
        this.ucName = cursor.getString(cursor.getColumnIndex(TableUCs.COLUMN_UC_NAME));
        this.districtCode = cursor.getString(cursor.getColumnIndex(TableUCs.COLUMN_DISTRICT_CODE));
        return this;
    }

    public String getUcCode() {
        return ucCode;
    }

    public void setUcCode(String uc_code) {
        this.ucCode = uc_code;
    }


    public String getUcName() {
        return ucName;
    }

    public void setUcName(String uc_name) {
        this.ucName = uc_name;
    }


    public String getDistrictCode() {
        return districtCode;
    }

    public void setDistrictCode(String districtCode) {
        this.districtCode = districtCode;
    }


    public JSONObject toJSONObject() throws JSONException {

        JSONObject json = new JSONObject();
        json.put(UCsContract.TableUCs.COLUMN_UC_CODE, this.ucCode == null ? JSONObject.NULL : this.ucCode);
        json.put(TableUCs.COLUMN_UC_NAME, this.ucName == null ? JSONObject.NULL : this.ucName);
        json.put(TableUCs.COLUMN_DISTRICT_CODE, this.districtCode == null ? JSONObject.NULL : this.districtCode);
        return json;
    }

}

File: app/src/main/java/edu/aku/hassannaqvi/casiregister/models/UCs.kt
package edu.aku.hassannaqvi.naunehal.models

import android.database.Cursor
import android.provider.BaseColumns
import org.json.JSONException
import org.json.JSONObject

/**
 * Created by hassan.naqvi on 10/31/2016.
 * @update ali.azaz
 */
class UCs {
    var ucCode: String = ""
    var ucName: String = ""
    var districtCode: String = ""

    @Throws(JSONException::class)
    fun sync(jsonObject: JSONObject): UCs {
        ucCode = jsonObject.getString(TableUCs.COLUMN_UC_CODE)
        ucName = jsonObject.getString(TableUCs.COLUMN_UC_NAME)
        districtCode = jsonObject.getString(TableUCs.COLUMN_DISTRICT_CODE)
        return this
    }

    fun hydrate(cursor: Cursor): UCs {
        ucCode = cursor.getString(cursor.getColumnIndex(TableUCs.COLUMN_UC_CODE))
        ucName = cursor.getString(cursor.getColumnIndex(TableUCs.COLUMN_UC_NAME))
        districtCode = cursor.getString(cursor.getColumnIndex(TableUCs.COLUMN_DISTRICT_CODE))
        return this
    }

    object TableUCs : BaseColumns {
        const val TABLE_NAME = "ucs"
        const val COLUMN_NAME_NULLABLE = "nullColumnHack"
        const val COLUMN_ID = "_ID"
        const val COLUMN_UC_CODE = "ucCode"
        const val COLUMN_UC_NAME = "ucName"
        const val COLUMN_DISTRICT_CODE = "districtCode"
    }
}